includes/admin_import.php

```diff
@@ -248,243 +371,280 @@ function tsml_import_page()
+			update_option('tsml_detection_test_mode', $tsml_detection_test_mode);
+			tsml_alert(__('Data Sources Debug Mode setting changed.', '12-step-meeting-list'));
+		}
+
+
 		/*debugging
 		delete_option('tsml_data_sources');
 		tsml_delete('everything');
 		tsml_delete_orphans();
 		*/
 ?>
 
-		<!-- Admin page content should all be inside .wrap -->
-		<div class="wrap">
+	<!-- Admin page content should all be inside .wrap -->
+	<div class="wrap ">
 
-			<h1></h1> <!-- Set alerts here -->
+		<h1></h1> <!-- Set alerts here -->
 
-			<div class="stack">
+		<div class="stack">
 
-				<?php if ($error) { ?>
-					<div class="error inline">
-						<p><?php echo $error ?></p>
-					</div>
-				<?php } elseif ($total = count($meetings)) { ?>
-					<div id="tsml_import_progress" class="progress" data-total="<?php echo $total ?>">
-						<div class="progress-bar"></div>
-					</div>
-					<ol id="tsml_import_errors" class="error inline hidden"></ol>
-				<?php } ?>
-
-				<!-- Import Data Sources -->
-				<div class="postbox stack">
-					<h2><?php _e('Import Data Sources', '12-step-meeting-list') ?></h2>
-					<p>
-						<?php printf(__('Data sources are JSON feeds that contain a website\'s public meeting data. They can be used to aggregate meetings from different sites into a single master list. 
-				Data sources listed below will pull meeting information into this website. A configurable schedule allows for each enabled data source to be scanned at least once per day looking 
-				for updates to the listing. Change Notification email addresses are sent an email when action is required to re-sync a data source with its meeting list information. 
-				Please note: records that you intend to maintain on your website should always be imported using the Import CSV feature below. <b>Data Source records will be overwritten when the 
-				parent data source is refreshed.</b> More information is available at the <a href="%s" target="_blank">Meeting Guide API Specification</a>.', '12-step-meeting-list'), 'https://github.com/code4recovery/spec') ?>
-					</p>
-					<?php if (!empty($tsml_data_sources)) { ?>
-						<table>
-							<thead>
-								<tr>
-									<th class="small align-center"></th>
-									<th><?php _e('Feed', '12-step-meeting-list') ?></th>
-									<th class="align-left"><?php _e('Parent Region', '12-step-meeting-list') ?></th>
-									<th class="align-left"><?php _e('Change Detection', '12-step-meeting-list') ?></th>
-									<th class="align-center"><?php _e('Meetings', '12-step-meeting-list') ?></th>
-									<th class="align-right"><?php _e('Last Refresh', '12-step-meeting-list') ?></th>
-									<th class="small"></th>
-								</tr>
-							</thead>
-							<tbody>
-								<?php foreach ($tsml_data_sources as $feed => $properties) { ?>
-									<tr data-source="<?php echo $feed ?>">
-										<td class="small ">
-											<form method="post" action="<?php echo $_SERVER['REQUEST_URI'] ?>">
```

Comment:
Mainly cosmetic, but I would wrap the form in a conditional so it doesn't render and can't be un-hidden in dev tools:

`<?php if ($properties['type'] !== 'CSV') { ?>
<form method="post" action="<?php echo $_SERVER['REQUEST_URI'] ?>">
												
											<?php wp_nonce_field($tsml_nonce, 'tsml_nonce', false) ?>
											<input type="hidden" name="tsml_add_data_source" value="<?php echo $feed ?>">
											<input type="hidden" name="tsml_add_data_source_name" value="<?php echo @$properties['name'] ?>">
											<input type="hidden" name="tsml_add_data_source_parent_region_id" value="<?php echo @$properties['parent_region_id'] ?>">
											<?php
												if($properties['type']!=='CSV')	{ ?>
													<input type="submit" value="Refresh" class="button button-small" style="display: block"; > 
											<?php } else { ?>  
													<input type="submit" value="Refresh" class="button button-small" style="display: none"; > 
											<?php } ?>													
										</form>
<?php } ?>`